src/factories/custom/Restricted_PIM_Factory_v1.sol

```diff
@@ -125,37 +131,65 @@ contract Restricted_PIM_Factory_v1 is
     }
 
     /// @inheritdoc IRestricted_PIM_Factory_v1
-    function addFunding(address beneficiary, address token, uint amount)
-        external
-    {
+    function addFunding(
+        address deployer,
+        address beneficiary,
+        address admin,
+        address token,
+        uint amount
+    ) external {
+        Funding storage funding = fundings[deployer][beneficiary][admin][token];
+        if (funding.sponsor != address(0) && funding.sponsor != _msgSender()) {
```

Comment:
This makes it possible to front run calls to this with 1 devimal of of the token and make adding funding fail